core/src/main/scala/org/bitcoins/core/wallet/utxo/AddressTag.scala

```diff
@@ -0,0 +1,74 @@
+package org.bitcoins.core.wallet.utxo
+
+sealed trait AddressTagType {
+  def tagType: String
+}
+
+sealed trait AddressTag {
+  def name: String
+  def tagType: String
+
+  def ==(at: AddressTag): Boolean = name == at.name && tagType == at.tagType
```

Comment:
nit: Should probably be `name == at.name && tagType.name == at.tagType.name`, that is unless you want to override equality checks in `AddressTagType` as well

Reply:
`tagType` is just a string

Reply:
Ah I see, shouldn't it be an `AddressTagType`?